pandas/core/groupby/groupby.py

```diff
@@ -1956,6 +1956,13 @@ def sum(
                 alias="add",
                 npfunc=np.sum,
             )
+        if self.axis == 1 and isinstance(self.obj.columns, MultiIndex):
```

Comment:
woa what is all this?

Reply:
we do not want to do things like this; should be handled much more generically.

Reply:
hi @jreback - made some changes. Let me know if this is fine.